### PR TITLE
LB-JE-ISSUE-THREE

### PR DESCRIPTION
**Pull request by AndrewEspejo**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### CODE REVIEW

File: src/components/List/tokenList.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  return (
+    <div>
+      {token}
+      <button onClick={handleNewToken()}></button>
+    </div>
+  );
+};
+/*
+const autentication = (tokenLocal) => {
```
Comment (alejandronanez) [inline src/components/List/tokenList.js]: Let's delete this commented out code from this file.

☝️ It's better to not leave any commented code in our codebase unless we're going to use it, soon. Which is rarely the case.
That said, if you really don't want to delete commented code, add a comment to it saying what is its purpose. Generally speaking, you don't need commented code at all.

File: src/components/List/tokenList.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export const AddToken = () => {
+  const tokenLocal = localStorage.getItem('token');
+  const [token, setToken] = useState(tokenLocal);
+
+  const handleNewToken = () => {
+    if (!tokenLocal) {
+      const newToken = getToken();
+      localStorage.setItem('token', newToken);
+    }
+    window.location.href = '/list';
```
Comment (alejandronanez) [inline src/components/List/tokenList.js:13]: 👍 This works, it's not the "react router way" of doing things, but gets the job done - nice work @lizhtml and @AndrewEspejo

👨‍🏫 In future implementations, you should use React Router's `useHistory` hook to navigate between pages.

```javascript
// Taken from React Router docs: https://reactrouter.com/web/api/Hooks/usehistory
import { useHistory } from "react-router-dom";

function HomeButton() {
  let history = useHistory();

  function handleClick() {
    // This is what's taking care of the navigation.
    history.push("/home");
  }

  return (
    <button type="button" onClick={handleClick}>
      Go home
    </button>
  );
}
```

File: src/Hooks/useLocalStorage.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+import getToken from '../lib/tokens';
+
+export const useLocalStorage = (key, initialValue) => {
```
Comment (alejandronanez) [inline src/Hooks/useLocalStorage.jsx:3]: 👍 I really like this abstraction, nice work 👏.

File: src/App.js [hunk 4]
```diff
@@ -1,23 +1,24 @@
 import React, { useState } from 'react';
 import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
 import './App.css';
-import Button from './components/Button';
+import TokenList from './Components/List/tokenList';
```
Comment (alejandronanez) [inline src/App.js]: This is causing an error in the build. Make sure you change `Components/` with `components` (lowercase) before merging.

**Review by alejandronanez — CHANGES_REQUESTED** [review]
@LizHtml @AndrewEspejo Nice work you did here, I love how you abstracted the token generation logic into a hook. Nicely done!

We have 2 things that we need to fix before merging!

1. You have a build error because React can't find the `tokenList` module - I left a comment about that.
2. The AC is not fully complete.

Per issue #3

> For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list

We don't have any button in the app that does this. Let's add this.

> Clicking the button/link generates a new token and saves it to localStorage

Because the button doesn't exist, this is not happening.

File: src/Hooks/useLocalStorage.jsx (hunk 3 above)
Comment (TitoAgudelo) [reply src/Hooks/useLocalStorage.jsx:3]: Great job!

File: src/App.js (hunk 4 above)
Comment (Lizhtml) [reply src/App.js]: Hello, Thank you very much for the comments, they help us to grow a lot. I want to say that when I try to rename "Components" it doesn't allow me
image
If I leave it in capital letters, the application runs normally
image
and if I put it in lowercase the compilation error occurs 
image

File: src/Hooks/useLocalStorage.jsx (hunk 3 above)
Comment (Lizhtml) [reply src/Hooks/useLocalStorage.jsx:3]: Thanks for checking it.

File: src/components/List/tokenList.js (hunk 2 above)
Comment (Lizhtml) [reply src/components/List/tokenList.js:13]: Thanks for the documentation it is very valuable to continue in our learning process.

File: src/Hooks/useLocalStorage.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      return initialValue;
+    }
+  });
+
+  const setValue = (value) => {
+    try {
+      setStoredValue(value);
+      window.localStorage.setItem(key, JSON.stringify(value));
```
Comment (DNeosphere) [inline src/Hooks/useLocalStorage.jsx:16]: ;3

File: src/components/List/tokenList.js (hunk 2 above)
Comment (DNeosphere) [reply src/components/List/tokenList.js:13]: overall, I like the architecture of tje project. Pretty nice job

Comment (DNeosphere) [reply src/components/List/tokenList.js:13]: This is not necessary, but I think is a good thing to learn and to have in mind [Protected Components](https://medium.com/@subalerts/creating-protected-routes-in-react-js-89e95974a822)

File: src/Hooks/useLocalStorage.jsx (hunk 5 above)
Comment (arche4) [inline src/Hooks/useLocalStorage.jsx:16]: I like it,  that could you explain me.... is great !!! 🥇 . Congratulations guys.

File: src/App.js (hunk 4 above)
Comment (alejandronanez) [reply src/App.js]: @Lizhtml @AndrewEspejo, it seems that error is gone now and the build is passing! 🤷 

Let's leave this as is then,

Thank you!

**Review by alejandronanez — APPROVED** [review]
@Lizhtml @AndrewEspejo nice work y'all 💪 

Let's make sure we make this change before merging.

- Move the `Create Token` button from the bottom nav to the homepage.
- Change `Create Token` to `Create List` or something like that, the idea is to present something more meaningful to the user

image


https://www.loom.com/share/ee5cdd0884624552bfdf9aa30e6772a2

**Pull request merged**